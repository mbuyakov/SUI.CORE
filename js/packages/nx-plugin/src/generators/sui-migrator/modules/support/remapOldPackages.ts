///////////////////////////////////
//FILE IS AUTOGENERATED. DON'T EDIT
///////////////////////////////////
export const importsByNewPackage = {
  "@sui/deps-amcharts": [
    "AMCHARTS",
    "AMEvent",
    "AOA2SheetOpts",
    "AutoFilterInfo",
    "BookType",
    "CellAddress",
    "CellObject",
    "ColInfo",
    "Comment",
    "Comments",
    "CommonOptions",
    "DateNFOption",
    "DefinedName",
    "ExcelDataType",
    "FullProperties",
    "getAmcharts",
    "getXlsx",
    "Hyperlink",
    "IRange",
    "JSON2SheetOpts",
    "ListTemplate",
    "MarginInfo",
    "NumberFormat",
    "OriginOption",
    "ParsingOptions",
    "Properties",
    "ProtectInfo",
    "Range",
    "RangeSpec",
    "RowInfo",
    "Sheet",
    "Sheet2CSVOpts",
    "Sheet2HTMLOpts",
    "Sheet2JSONOpts",
    "SheetAOAOpts",
    "SheetJSONOpts",
    "SheetKeys",
    "SheetProps",
    "SheetType",
    "StreamUtils",
    "StrictWS",
    "Table2SheetOpts",
    "WBProps",
    "WBView",
    "WorkBook",
    "WorkbookProperties",
    "WorkSheet",
    "WritingOptions",
    "WSKeys",
    "WSSpec",
    "XLSX",
    "XLSX$Consts",
    "XLSX$Utils"
  ],
  "@sui/deps-antd": [
    "ArgsProps",
    "ButtonGroupProps",
    "ButtonHTMLType",
    "ButtonType",
    "Calendar",
    "CardType",
    "CheckboxChangeEvent",
    "ColumnType",
    "CustomIconComponentProps",
    "DatePicker",
    "DatePickerProps",
    "FormListProps",
    "NamePath",
    "PresetColorType",
    "RangePickerProps",
    "RcFile",
    "ruRU",
    "SelectValue",
    "SizeType",
    "TextAreaProps",
    "ThemeConfig"
  ],
  "@sui/deps-ioc": [
    "Container",
    "OnlyInstantiableByContainer",
    "Scope",
    "Singleton",
    "SingletonAndOnlyIoc"
  ],
  "@sui/deps-material": [
    "bindHover",
    "bindMenu",
    "HoverMenu",
    "PopupState",
    "TransitionProps",
    "usePopupState"
  ],
  "@sui/lib-auth": [
    "AccessRights",
    "Allowed",
    "AuthorityList",
    "checkAuthority",
    "getPrivileges",
    "getUser",
    "hasAnyPrivilege",
    "hasPrivilege",
    "hasRole",
    "ICoreUser",
    "isAdmin",
    "LibAuthModule",
    "Privilege",
    "SYSTEM_USER_ID",
    "UserNotInitializedError",
    "UserService",
    "UserServiceImpl"
  ],
  "@sui/lib-cache-abstract": [
    "CacheEntryUpdatedEvent",
    "CacheManager",
    "ICacheEntry"
  ],
  "@sui/lib-color": [
    "Color",
    "ColorHeatMap",
    "ColorHeatMapImpl",
    "findColorBetween",
    "IColorHeatMapSettings",
    "LibColorModule"
  ],
  "@sui/lib-event-manager": [
    "EventListenerDescriptor",
    "EventManager",
    "SuiEvent",
    "SuiEventListener"
  ],
  "@sui/lib-hooks": [
    "useAsyncEffect",
    "useAsyncState",
    "useHandler",
    "usePromise",
    "UsePromiseState",
    "useService",
    "useValue"
  ],
  "@sui/lib-module-manager": [
    "MainModule",
    "ModuleDepsInitEvent",
    "ModuleDepsInitializedEvent",
    "ModuleDiscoveredEvent",
    "ModuleDuplicateEvent",
    "ModuleErrorEvent",
    "ModuleEvent",
    "ModuleFailedEvent",
    "ModuleInitEvent",
    "ModuleInitializedEvent",
    "ModuleManager",
    "ModuleWithDurationEvent",
    "SuiModule",
    "SuiModuleWithSettings"
  ],
  "@sui/lib-notification-dispatcher": [
    "ErrorEvent",
    "LibNotificationDispatcherModule",
    "NotificationArgsProps",
    "NotificationCloseEvent",
    "NotificationDispatcher",
    "NotificationDispatcherImpl",
    "NotificationEvent",
    "NotificationSeverity"
  ],
  "@sui/lib-storage": [
    "LibStorageModule",
    "LocalStorageService",
    "LocalStorageServiceImpl",
    "LSKeyWrapper",
    "LSKeyWrapperImpl"
  ],
  "@sui/lib-typescript-ast": [
    "astQuery",
    "astRemove",
    "astReplace",
    "ImportWithAlias",
    "ParsedImportDeclaration",
    "parseImportDeclaration",
    "parseImportSpecifier",
    "printImportDeclaration",
    "printNode",
    "printNodes"
  ],
  "@sui/ui-actions": [
    "Button",
    "ButtonProps",
    "IOnClickIntegration",
    "IUsePopconfirmState",
    "PopconfirmSettings",
    "useOnClick",
    "UseOnClickState",
    "usePopconfirm"
  ],
  "@sui/ui-charts": [
    "CategoryColumnChart",
    "ChartWrapper",
    "ChordDiagramWrapper",
    "ForceDirectedTree",
    "ForceDirectedTreeWrapper",
    "IAdditionalSettingProps",
    "ICategoryColumnChartSeries",
    "IChartWrapperProps",
    "IForceDirectedTreeAdditionalSettingProps",
    "IForceDirectedTreeDataElement",
    "IReportTopBarChartSettingProps",
    "MapWrapper",
    "PieChart3DWrapper",
    "PieChartWrapper",
    "RadarChartWrapper",
    "ReportTopBarChart",
    "SlicedChartWrapper",
    "TreeMapWrapper",
    "WordCloudWrapper",
    "XYChart3DWrapper",
    "XYChartWrapper"
  ],
  "@sui/ui-idle-tracker": [
    "IdleTimerConsumer",
    "IdleTracker",
    "IIdleTimer",
    "UiIdleTrackerModule",
    "useIdleTimerContext"
  ],
  "@sui/ui-layout": [
    "Exception403",
    "Exception404"
  ],
  "@sui/ui-material": [
    "ProgressIcon",
    "ProgressIconProps",
    "TooltipWrapper",
    "TooltipWrapperProps"
  ],
  "@sui/ui-module-manager": [
    "DisablableModuleSettings",
    "ModuleManagerRoot",
    "ModulesSettings"
  ],
  "@sui/ui-notification-handler": [
    "NotificationViewer",
    "UiNotificationHandlerModule"
  ],
  "@sui/ui-observable": [
    "IObservableBinderProps",
    "joinObservables2",
    "joinObservables3",
    "joinObservables4",
    "joinObservables5",
    "joinObservables6",
    "Observable",
    "ObservableBinder",
    "ObservableHandler",
    "ObservableHandlerStub",
    "ObservableLocalStorageValue",
    "useObservableSubscribe",
    "useObservableValue"
  ],
  "@sui/ui-themes": [
    "antdDefaultTheme",
    "AntdTheme",
    "compileFinalTheme",
    "getComponents",
    "ISuiThemeContext",
    "KludgeForStorybook",
    "mergeSuiComponentThemes",
    "muiDefaultTheme",
    "MuiTheme",
    "SuiComponentsTheme",
    "SuiComponentsThemeCompiled",
    "SuiComponentsThemeConfig",
    "SuiComponentTheme",
    "SuiComponentThemeCompiled",
    "SuiComponentThemeConfig",
    "suiDefaultTheme",
    "SuiTheme",
    "SuiThemeCompiled",
    "SuiThemeComponents",
    "SuiThemeConfig",
    "SuiThemeContext",
    "SuiThemeProvider",
    "SuiThemeVariants",
    "ThemeChangedEvent",
    "ThemeService",
    "ThemeServiceImpl",
    "ThemeSwitchButton",
    "UiThemesModule"
  ],
  "@sui/util-chore": [
    "addCss",
    "clamp",
    "concatDataKey",
    "DataKey",
    "DataKeyNode",
    "dataKeysToDataTree",
    "getDataByKey",
    "getTOrCall",
    "lazyInit",
    "normalizeDataKey",
    "sleep",
    "TOrCallback"
  ],
  "@sui/util-types": [
    "Class",
    "IObjectWithIndex",
    "NotFunction",
    "Nullable"
  ]
};
