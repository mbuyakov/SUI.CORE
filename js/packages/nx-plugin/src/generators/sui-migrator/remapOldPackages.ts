import {tsquery} from "@phenomnomnominal/tsquery";
import {factory, ImportDeclaration, ImportSpecifier, isNamedImports, StringLiteral} from "typescript";
import {printNode} from "./util";
import {logWithPrefix} from "../../utils/logger";

const newImports = {
  "@sui/ui-observable": [
    "Observable",
    "IObservableBinderProps",
    "ObservableBinder",
    "ObservableLocalStorageValue",
    "ObservableHandler",
    "ObservableHandlerStub",
    "joinObservables2",
    "joinObservables3",
    "joinObservables4",
    "joinObservables5",
    "joinObservables6"
  ],
  "@sui/util-chore": [
    "DataKey",
    "normalizeDataKey",
    "concatDataKey",
    "getDataByKey",
    "DataKeyNode",
    "dataKeysToDataTree",
    "TOrCallback",
    "getTOrCall"
  ],
  "@sui/util-types": [
    "Class",
    "Nullable",
    "NotFunction",
    "IObjectWithIndex"
  ]
};

const inversedNewImports = Object.keys(newImports)
  .flatMap(key => newImports[key].map(value => [value, key]))
  .reduce((prev, cur) => {
    prev[cur[0]] = cur[1];
    return prev;
  }, {});

export function remapOldPackages(content: string): string {

  return tsquery.replace(content, "ImportDeclaration", (node: ImportDeclaration) => {
    const moduleName = (node.moduleSpecifier as StringLiteral).text;

    if (!(["@sui/all", "@sui/ui-old-core", "@sui/ui-old-react"].includes(moduleName) || moduleName.startsWith("@/"))) {
      return;
    }

    if (!isNamedImports(node.importClause.namedBindings)){
      return;
    }

    let importSpecifiers: ImportSpecifier[] = [...node.importClause.namedBindings.elements];

    if (importSpecifiers.every(it => !inversedNewImports[it.name.text])) {
      return;
    }

    const imports: ImportDeclaration[] = [];

    importSpecifiers = importSpecifiers.filter(it => {
      if (inversedNewImports[it.name.text]) {
        logWithPrefix("remapOldPackages", `Replace ${it.name.text} from ${moduleName} to ${inversedNewImports[it.name.text]}`);

        imports.push(
          factory.createImportDeclaration(
            undefined,
            factory.createImportClause(
              false,
              undefined,
              factory.createNamedImports([
                factory.createImportSpecifier(
                  false,
                  undefined,
                  factory.createIdentifier(it.name.text)
                )
              ])
            ),
            factory.createStringLiteral(inversedNewImports[it.name.text])
          )
        );

        return false;
      } else {
        return true;
      }
    });

    if (importSpecifiers.length) {
      imports.push(
        factory.createImportDeclaration(
          undefined,
          factory.createImportClause(
            false,
            undefined,
            factory.createNamedImports(importSpecifiers)
          ),
          factory.createStringLiteral(moduleName)
        )
      );
    }

    return imports.map(printNode)
      .join("\n")
      .replace(/{ /g, "{")
      .replace(/ }/g, "}");
  });
}
