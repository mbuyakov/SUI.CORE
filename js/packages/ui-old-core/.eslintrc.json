{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ],
  "rules": {
    "@typescript-eslint/no-inferrable-types": 0,
    "no-useless-escape": 0,
    "eqeqeq": 0,
    "no-case-declarations": 0,
    "@typescript-eslint/no-explicit-any": 0,
    "no-extra-boolean-cast": 0,
    "prefer-const": 0,
    "no-control-regex": 0,
    "@typescript-eslint/ban-ts-comment": 0
  }
}
