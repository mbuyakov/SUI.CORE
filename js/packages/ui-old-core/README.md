# ui-old-core

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test ui-old-core` to execute the unit tests via [Jest](https://jestjs.io).
