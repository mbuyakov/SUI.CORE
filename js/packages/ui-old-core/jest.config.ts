/* eslint-disable */
export default {
  displayName: "ui-old-core",
  preset: "../../jest.preset.js",
  transform: {
    "^(?!.*\\.(js|jsx|ts|tsx|css|json)$)": "@nx/react/plugins/jest",
    "^.+\\.[tj]sx?$": "babel-jest",
  },
  moduleFileExtensions: ["ts", "tsx", "js", "jsx"],
  coverageDirectory: "../../coverage/packages/ui-old-core",
};
