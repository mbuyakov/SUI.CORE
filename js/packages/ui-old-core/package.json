{
  "name": "@sui/ui-old-core",
  "version": "0.0.1",
  "dependencies": {
    "@stomp/stompjs": "^5.4.4",
    "apollo-cache-inmemory": "^1.6.5",
    "apollo-client": "^2.6.8",
    "apollo-link": "^1.2.14",
    "apollo-link-context": "^1.0.20",
    "apollo-link-http": "^1.5.17",
    "async-validator": "3.5.1",
    "autobind-decorator": "^2.4.0",
    "axios": "^0.19.2",
    "graphql": "^14.6.0",
    "graphql-tag": "^2.10.3",
    "jsonpack": "^1.1.5",
    "jszip": "^3.5.0",
    "lodash": "^4.17.15",
    "pako": "^1.0.11",
    "moment": "2.29.4"
  }
}
