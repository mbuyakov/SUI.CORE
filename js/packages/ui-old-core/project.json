{
  "name": "ui-old-core",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/ui-old-core/src",
  "projectType": "library",
  "tags": ["type:ui"],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/ui-old-core/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/ui-old-core",
        "tsConfig": "packages/ui-old-core/tsconfig.lib.json",
        "project": "packages/ui-old-core/package.json",
        "entryFile": "packages/ui-old-core/src/index.ts",
        "external": ["@emotion/react/jsx-runtime"],
        "rollupConfig": "@nx/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "packages/ui-old-core/README.md",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    }
  }
}
