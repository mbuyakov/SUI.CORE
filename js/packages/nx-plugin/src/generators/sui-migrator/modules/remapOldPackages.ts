import {ImportDeclaration} from "typescript";
import {logWithPrefix} from "../../../utils/logger";
import {
  astReplace,
  ParsedImportDeclaration,
  parseImportDeclaration,
  printImportDeclaration,
  printNodes
} from "@sui/lib-typescript-ast";
import {importsByNewPackage} from "./support/remapOldPackages";
import {oldPackages} from "../../../utils/oldPackages";


const newPackageByImports = Object.keys(importsByNewPackage)
  .flatMap(key => importsByNewPackage[key].map(value => [value, key]))
  .reduce((prev, cur) => {
    prev[cur[0]] = cur[1];
    return prev;
  }, {});

export function remapOldPackages(packageName: string, content: string): string {
  const isOldPackage = oldPackages.includes(packageName);

  return astReplace(content, "ImportDeclaration", (node: ImportDeclaration) => {
    const parsedImport = parseImportDeclaration(node);
    const isImportFromOldModule = oldPackages.includes(parsedImport.from);

    if (
      !isImportFromOldModule
      // for internal SUI packages
      && !(isOldPackage && parsedImport.from.startsWith("@/"))) {
      return;
    }

    if (!parsedImport.namedImports.length) {
      return;
    }

    let namedImports = [...parsedImport.namedImports];
    if (namedImports.every(it => !newPackageByImports[it.originalName])) {
      return;
    }

    const imports: ParsedImportDeclaration[] = [];

    namedImports = namedImports.filter(it => {
      const newPackageByImport = newPackageByImports[it.originalName];
      if (newPackageByImport) {
        logWithPrefix("remapOldPackages", `Replace ${it.originalName} from ${parsedImport.from} to ${newPackageByImport}`);

        imports.push({
          namedImports: [
            {
              originalName: it.originalName,
              alias: it.alias
            }
          ],
          from: newPackageByImport
        });

        return false;
      } else {
        return true;
      }
    });

    if (namedImports.length) {
      imports.push({
        namedImports,
        from: parsedImport.from
      });
    }

    return printNodes(imports.map(it => printImportDeclaration(it)));
  });
}
