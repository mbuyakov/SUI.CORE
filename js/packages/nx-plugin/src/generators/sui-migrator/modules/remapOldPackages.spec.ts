import {remapOldPackages} from "./remapOldPackages";

describe("remap old imports", () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should replace old import", () => {
    const consoleLog = jest.spyOn(console, "log");
    expect(remapOldPackages("@sui/deps-antd", `
import {DataKey, Test} from "@sui/ui-old-react";
import {dataKeysToDataTree} from "@sui/ui-old-core";
    `.trim())).toMatchSnapshot();
    expect(consoleLog).toBeCalledTimes(2);
  });
});
