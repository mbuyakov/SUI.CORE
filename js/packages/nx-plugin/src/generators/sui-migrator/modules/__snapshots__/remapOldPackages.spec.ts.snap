// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remap old imports should replace old import 1`] = `
"import {DataKey} from \\"@sui/util-chore\\";
import {Test} from \\"@sui/ui-old-react\\";
import {dataKeysToDataTree} from \\"@sui/util-chore\\";"
`;
